probes/sastToolRunsOnAllCommits/impl.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package sastToolRunsOnAllCommits
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const Probe = "sastToolRunsOnAllCommits" //#nosec
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	r := raw.SASTResults
+
+	f, err := finding.NewWith(fs, Probe,
+		"", nil,
+		finding.OutcomePositive)
+	if err != nil {
+		return nil, Probe, fmt.Errorf("create finding: %w", err)
+	}
+
+	totalMerged := len(r.Commits)
+	totalTested := 0
+
+	for i := range r.Commits {
+		wf := &r.Commits[i]
+		if wf.Compliant {
+			totalTested++
+		}
+	}
+
+	if totalMerged == 0 {
+		f = f.WithOutcome(finding.OutcomeNotApplicable)
+		f = f.WithMessage("no pull requests merged into dev branch")
+		return []finding.Finding{*f}, Probe, nil
+	}
+
+	f = f.WithValue("totalTested", totalTested)
+	f = f.WithValue("totalMerged", totalMerged)
+
+	if totalTested == totalMerged {
+		f = f.WithOutcome(finding.OutcomePositive)
```

Comment:
nit: you can simplify the code a little:
```go
f = f.WithOutcome(finding.OutcomePositive).WithMessage(...)
```

same comment below

Reply:
Done in https://github.com/ossf/scorecard/pull/3571/commits/c25e04612ad6313a5d27f49f6d9f2fdfe75cbae7